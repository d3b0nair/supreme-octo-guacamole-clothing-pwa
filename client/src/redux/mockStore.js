import { createStore } from "redux";
import { persistReducer } from "redux-persist";
import { rootReducer, persistConfig } from "./root-reducer";
export const createMockStore = ({ mockState, mockReducers }) => {
  const combinedReducers = persistReducer(
    persistConfig,
    rootReducer(mockReducers)
  );
  return createStore(combinedReducers, mockState);
};
