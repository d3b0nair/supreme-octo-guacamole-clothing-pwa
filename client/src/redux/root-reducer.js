import { combineReducers } from "redux";
import { persistReducer } from "redux-persist";
import storage from "redux-persist/lib/storage";

import userReducer from "./user/user.reducer";
import cartReducer from "./cart/cart.reducer";
import directoryReducer from "./directory/directory.reducer";
import shopReducer from "./shop/shop.reducer";
import checkoutReducer from "./checkout/checkout.reducer";
import stripeReducer from "./stripe/stripe.reducer";
import orderReducer from "./order/order.reducer"

const persistConfig = {
  key: "root",
  storage,
  whitelist: ["cart", "stripe"],
};
const rootReducer = combineReducers({
  user: userReducer,
  cart: cartReducer,
  checkout: checkoutReducer,
  directory: directoryReducer,
  shop: shopReducer,
  stripe: stripeReducer,
  selectedOrder: orderReducer
});
export default persistReducer(persistConfig, rootReducer);
