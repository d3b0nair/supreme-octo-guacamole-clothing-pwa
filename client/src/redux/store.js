import { createStore, applyMiddleware, compose } from "redux";
import { persistStore } from "redux-persist";
import logger from "redux-logger";
import createSagaMiddleware from "redux-saga";

import rootReducer from "./root-reducer";
import rootSaga from "./root-saga";

const sagaMiddleware = createSagaMiddleware();

const middlewares = [sagaMiddleware];
const composeEnhancers =
  (process.env.NODE_ENV !== "production" &&
    typeof window !== "undefined" &&
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) ||
  compose;
if (process.env.NODE_ENV === "development") {
  middlewares.push(logger);
}
export const store = createStore(
  rootReducer,
  composeEnhancers(applyMiddleware(...middlewares))
);

sagaMiddleware.run(rootSaga);

export const persistor = persistStore(store);
