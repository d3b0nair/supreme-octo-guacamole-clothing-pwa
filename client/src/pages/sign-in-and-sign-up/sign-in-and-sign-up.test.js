import React from "react";
import { render, cleanup } from "@testing-library/react";
import SignInAndSignUpPage from "./sign-in-and-sign-up.component";
import { createMockStore } from "../../redux/mockStore";

import { Provider } from "react-redux";

describe("SignIn and SignUp Page", () => {
  afterEach(cleanup);
  const component = () => (
    <Provider store={store}>
      <SignInAndSignUpPage />
    </Provider>
  );
  const mockReducer = (state = {}, action) => state;
  const store = createMockStore({
    state: {},
    reducers: { testReducer: mockReducer },
  });
  const { container } = render(component());

  it("should mount", () => {
    jest.spyOn(window.screen, "width", "get").mockReturnValue(800);
    jest.spyOn(window.screen, "height", "get").mockReturnValue(800);
    expect(container).toMatchSnapshot();
  });
});
