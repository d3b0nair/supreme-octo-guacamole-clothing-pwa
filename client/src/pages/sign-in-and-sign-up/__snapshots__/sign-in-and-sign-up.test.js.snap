// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn and SignUp Page should mount 1`] = `
<div>
  <div
    class="sc-egiyK gqCjcp"
  >
    <div
      class="sc-eCImPb kiJsgT"
    >
      <h2
        class="sc-jRQBWg hRSpAi"
      >
        I already have an account
      </h2>
      <span>
        Sign in with your email and password
      </span>
      <form>
        <div
          class="sc-bdvvtL hMBGla"
        >
          <input
            class="sc-gsDKAQ hVBime"
            data-testid="FormInputContainer"
            name="email"
            required=""
            type="email"
          />
          <label
            class="sc-dkPtRN jpMQjL  form-input-label"
            data-testid="FormInputLabel"
          >
            E-Mail
          </label>
        </div>
        <div
          class="sc-bdvvtL hMBGla"
        >
          <input
            class="sc-gsDKAQ hVBime"
            data-testid="FormInputContainer"
            name="password"
            required=""
            type="password"
          />
          <label
            class="sc-dkPtRN jpMQjL  form-input-label"
            data-testid="FormInputLabel"
          >
            Password
          </label>
        </div>
        <div
          class="sc-gKclnd idUVbs"
        >
          <button
            class="sc-hKwDye hzDzeB"
            type="submit"
          >
            Sign in
          </button>
          <button
            class="sc-hKwDye hPNorZ"
            data-testid="CustomButtonForGoogleSignInStart"
            type="button"
          >
            Sign in with Google
          </button>
        </div>
      </form>
    </div>
    <div
      class="sc-iCfMLu cUjxrA"
    >
      <h2
        class="sc-kDTinF laMaXn"
      >
        I do not have an account
      </h2>
      <span>
        Sign up with your email and password
      </span>
      <form
        class="sign-up-form"
      >
        <div
          class="sc-bdvvtL hMBGla"
        >
          <input
            class="sc-gsDKAQ hVBime"
            data-testid="displayName"
            name="displayName"
            required=""
            type="text"
          />
          <label
            class="sc-dkPtRN jpMQjL  form-input-label"
            data-testid="FormInputLabel"
          >
            Display Name
          </label>
        </div>
        <div
          class="sc-jrQzAO gWnczy"
        >
          <div
            class="sc-bdvvtL hMBGla"
          >
            <input
              class="sc-gsDKAQ hVBime"
              data-testid="email"
              name="email"
              required=""
              type="email"
            />
            <label
              class="sc-dkPtRN jpMQjL  form-input-label"
              data-testid="FormInputLabel"
            >
              Email
            </label>
          </div>
        </div>
        <div
          class="sc-jrQzAO gWnczy"
        >
          <div
            class="sc-bdvvtL hMBGla"
          >
            <input
              class="sc-gsDKAQ hVBime"
              data-testid="password"
              name="password"
              pattern="^(?=.*)(?=.*[a-z])(?=.*[A-Z])(?=.*[a-zA-Z]).{8,}$"
              required=""
              type="password"
            />
            <label
              class="sc-dkPtRN jpMQjL  form-input-label"
              data-testid="FormInputLabel"
            >
              Password
            </label>
          </div>
          <svg
            class="sc-pVTFL illzAJ"
          >
            hidden.svg
          </svg>
        </div>
        <div
          class="sc-jrQzAO gWnczy"
        >
          <div
            class="sc-bdvvtL hMBGla"
          >
            <input
              class="sc-gsDKAQ hVBime"
              data-testid="confirmPassword"
              name="confirmPassword"
              required=""
              type="password"
            />
            <label
              class="sc-dkPtRN jpMQjL  form-input-label"
              data-testid="FormInputLabel"
            >
              Confirm Password
            </label>
          </div>
          <svg
            class="sc-pVTFL illzAJ"
          >
            hidden.svg
          </svg>
        </div>
        <button
          class="sc-hKwDye hzDzeB"
          data-testid="CustomButtonForSubmit"
          style="width: 100%;"
          type="submit"
        >
          Sign up
        </button>
      </form>
    </div>
  </div>
</div>
`;
